### PR TITLE
LibWasm: Change dir_fd to auto instead of int

### PR DESCRIPTION
**Pull request by nipos**
Solaris defines AT_FDCWD as 0xffd19553, which makes it an unsigned int.
Defining dir_fd as int makes it complain about comparing an int with an unsigned int later when AT_FDCWD and dir_fd are compared.
Defining dir_fd as auto makes it an unsigned int on Solaris and solves the issue.
On other platforms, I don't expect any noticeable change by this, so I decided to just replace it and not use an ifdef.
Solaris' AT_FDCWD definition seems quite unusual, which causes problems also in other projects, see https://bugs.python.org/issue15965 for example.

### CODE REVIEW

**Comment by alimpfard** [comment]
:thinking: a bit weird, does that mean solaris has this compiling with a warning?
```c++
auto fd = AT_FDCWD;
openat(fd, "foo", 0); // warning: implicit unsigned -> int conversion
```

**Comment by nipos** [comment]
That snipped you gave fails for other reasons,and I'm not good enough at C++ to fully understand that:
```
nipos@openindiana ~ [1]> clang test.c
test.c:8:12: error: expected identifier
openat(fd, "foo", 0);
           ^
test.c:8:1: warning: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
openat(fd, "foo", 0);
^
int
test.c:8:1: warning: a function declaration without a prototype is deprecated in all versions of C and is treated as a zero-parameter prototype in C2x, conflicting with a previous declaration [-Wdeprecated-non-prototype]
/usr/include/fcntl.h:140:12: note: conflicting prototype is here
extern int openat(int, const char *, int, ...);
           ^
test.c:8:1: error: conflicting types for 'openat'
openat(fd, "foo", 0);
^
/usr/include/fcntl.h:140:12: note: previous declaration is here
extern int openat(int, const char *, int, ...);
           ^
2 warnings and 2 errors generated.
```
The error while compiling SerenityOS was at the line `if (dir_fd < 0 && dir_fd != AT_FDCWD)` because dir_fd (declared as int) is compared with AT_FDCWD (which is an unsigned it).

**Comment by alimpfard** [comment]
😅 that sample was supposed to go inside a function, but nevermind that.

**Comment by nipos** [comment]
I tried it again within a function but I get unexpected results.
It works perfectly fine if I define fd as int and also if I define it as unsigned int,but if I define it as auto,it throws the following warning:
```
nipos@openindiana ~> clang test.c
test.c:8:6: warning: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
auto fd = AT_FDCWD;
~~~~ ^
int
1 warning generated.
```

**Comment by BertalanD** [comment]
Looks like you're compiling it as C, not C++. In pre-C23 C, `auto` just means that the variable is allocated on the stack, it doesn't do type inference like in C++, you'd still need to explicitly specify the type.

tl;dr: compile your test file as C++

**Comment by alimpfard** [comment]
Anyway, I think your patch is okay to merge, so let's get it merged
